src/main/java/com/intuit/fuzzymatcher/component/TokenRepo.java

```diff
@@ -46,6 +47,9 @@ public void put(Token token) {
 
         TreeSet<Object> tokenBinaryTree;
 
+        private static final double MAX_AGE_DIFF = 5;
+        private static final double MIN_AGE_DIFF = 1;
+
```

Comment:
Instead of hard coding the range values, was looking for this to be changed by the user. Currently users have the ability to change neighborhoodRange value in Element
https://github.com/intuit/fuzzy-matcher/blob/d2ce6f6f53a2ea5b1d628bd2fb0aec5d1d22bc5a/src/main/java/com/intuit/fuzzymatcher/domain/Element.java#L171

By default it is set to 0.9, with this default if we have a Number of 10.0, all other values within the range 9.0-11.0 will give a match.
Users have the ability to increase this to 0.95 and then the range will be limited to 9.5-10.5

On similar lines, wanted AGE element to give a match within 1 year, when the default neighborhoodRange is 0.9 , and as users change increase it the range will narrow. But as users decrease it the range will widen until 5 years

The 1 and 5 year values in the issue are just approximate scale within which the range can be varied .
If you can think of a mathematical formula that can change the neighborhoodRange that user can input and internally vary it.

Reply:
@manishobhatia The `neighborhoodRange` is still honored. This is indeed a new function which is still (partially) linear but capped at ends (viz 1 and 5 in the case of AGE). `neighborhoodRange` can still be used to widen or narrow the range as follows:

Reply:
thanks @tdjogi010 . This makes more sense now.
In my mind, I was thinking that the range is linear to neighorhoodRange value, rather than the AGE value.

I understand the implementation for NUMBER is linear to its value. I made a mistake thinking that AGE can be implemented in the same manner.

Since the NUMBER can have a very wide values, for example someone trying to match money transaction in $10's will care about similarity within $1 , but if someone is matching values in $1000's will have a more relaxed similarity requirement with 10's or 100's

With AGE the variation is not so much, the values will mostly be within 100 years. So I think we can implement it a little different. 

Here is the Math in my mind
range = (1 - neighborHoodRange) * 10

So for Age = 10
neighborHoodRange of 0.9 we will get a range of 9-11
neighborHoodRange of 0.8 we will get a range of 8-12

And Age = 50 will have a similar absolute value
neighborHoodRange of 0.9 we will get a range of 49-51
neighborHoodRange of 0.8 we will get a range of 48-52

This might be more intuitive to the users. Let me know if this makes sense.

Reply:
@manishobhatia It seems perfectly fine to use it with age since it is a very valid assumption for AGE to be below 100. Small caveat is that range can go beyond 5 with the approach you suggested whereas #36 says to cap it beyond 5. As long as capping is not a strict requirement, I can go ahead with it. Let me know if its good to implement.

Reply:
yes capping is not a hard requirement. It is best to leave the controls with the end user, and not have it hard coded. Thanks @tdjogi010